client/src/modules/vouchers/complex-voucher.ctrl.js

```diff
@@ -105,6 +111,134 @@ function ComplexJournalVoucherController(Vouchers, $translate, Currencies, Sessi
       });
   }
 
+  // open generic income function
+  function openGenericIncomeTool() {
```

Comment:
It seems to me that all of these tool chains do exactly the same thing which is to configure the rows - we could greatly reduce the code here by making all of these one function. 

For example: 
```js
Toolkit.open(voucherTool)
  .then(processVoucherToolRows);

function processVoucherToolRows(result) { 
  // ... 
}
```

We also then may not even need the opening methods - I'm not sure exactly how this would work: 
```js
genericIncomeTool.action = openVoucherTool(genericIncomeTool);

function openVoucherTool(voucherTool) { 
  return function() { 
    Toolkit.open(voucherTool) //...
  }
}
```

There are a lot of patterns here that could be reduced into helper methods with clear comments!